File: app/src/main/java/ca/pkay/rcloneexplorer/Items/Task.java
package ca.pkay.rcloneexplorer.Items;

public class Task {

    public static String TABLE_NAME = "task_table";

    public static String COLUMN_NAME_ID= "task_id";
    public static String COLUMN_NAME_TITLE = "task_title";
    public static String COLUMN_NAME_REMOTE_ID = "task_remote_id";
    public static String COLUMN_NAME_REMOTE_TYPE = "task_remote_type";
    public static String COLUMN_NAME_REMOTE_PATH = "task_remote_path";
    public static String COLUMN_NAME_LOCAL_PATH = "task_local_path";
    public static String COLUMN_NAME_SYNC_DIRECTION = "task_direction";

    private Long id;

    private String title="";
    private String remoteId ="";
    private int remoteType =0;
    private String remotePath ="";
    private String localPath = "";
    private int direction=0;


    public Task(Long id) {
        this.id = id;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getRemoteId() {
        return remoteId;
    }

    public void setRemoteId(String remoteId) {
        this.remoteId = remoteId;
    }

    public int getRemoteType() {
        return remoteType;
    }

    public void setRemoteType(int remoteType) {
        this.remoteType = remoteType;
    }

    public String getRemotePath() {
        return remotePath;
    }

    public void setRemotePath(String remotePath) {
        this.remotePath = remotePath;
    }

    public String getLocalPath() {
        return localPath;
    }

    public void setLocalPath(String localPath) {
        this.localPath = localPath;
    }

    public int getDirection() {
        return direction;
    }

    public void setDirection(int direction) {
        this.direction = direction;
    }

    public String toString(){
        return title + ": " + remoteId + ": " + remoteType + ": " + remotePath + ": " + localPath + ": " + direction;
    }
}

File: app/src/main/java/ca/pkay/rcloneexplorer/Items/Task.kt
package ca.pkay.rcloneexplorer.Items

import kotlinx.serialization.Serializable
import kotlinx.serialization.encodeToString
import kotlinx.serialization.json.Json
import kotlinx.serialization.json.JsonNames
import org.json.JSONObject

@Serializable
data class Task(var id: Long) {
    // Alternatives are kept for backwards compatibility with old, manual parser
    @JsonNames("name") var title = ""
    var remoteId = ""
    var remoteType = 0
    var remotePath = ""
    var localPath = ""
    @JsonNames("syncDirection") var direction = 0
    var md5sum = TASK_MD5SUM_DEFAULT
    var wifionly = TASK_WIFIONLY_DEFAULT

    override fun toString(): String {
        return "$title: $remoteId: $remoteType: $remotePath: $localPath: $direction"
    }

    companion object {
        var TABLE_NAME = "task_table"
        var COLUMN_NAME_ID = "task_id"
        var COLUMN_NAME_TITLE = "task_title"
        var COLUMN_NAME_REMOTE_ID = "task_remote_id"
        var COLUMN_NAME_REMOTE_TYPE = "task_remote_type"
        var COLUMN_NAME_REMOTE_PATH = "task_remote_path"
        var COLUMN_NAME_LOCAL_PATH = "task_local_path"
        var COLUMN_NAME_SYNC_DIRECTION = "task_direction"
        var COLUMN_NAME_MD5SUM = "task_use_md5sum"
        var COLUMN_NAME_WIFI_ONLY = "task_use_only_wifi"

        const val TASK_MD5SUM_DEFAULT = false
        const val TASK_WIFIONLY_DEFAULT = false

        fun fromString(json: String): Task {
            return Json.decodeFromString(json)
        }
    }

    fun asJSON(): JSONObject {
        return JSONObject(Json.encodeToString(this))
    }
}